core/src/main/java/io/questdb/mp/RingQueue.java

```diff
@@ -35,6 +35,7 @@
 
     @SuppressWarnings("unchecked")
     public RingQueue(ObjectFactory<T> factory, int cycle) {
+        assert cycle < 3 || cycle == (cycle / 2) * 2;
```

Comment:
I don't think this is correct to check that cycle is power of 2. Should be something like `Integer.bitCount(cycle) < 2`

Reply:
(cycle & (cycle - 1) == 0, if cycle != 0

Reply:
Thank you champs 👍